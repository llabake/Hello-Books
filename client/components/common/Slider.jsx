import React, { Component } from 'react';



import sliderimage1 from '../../media/archive-beautiful-book-stack-256455.jpg';
import sliderimage2 from '../../media/bookcase-books-bookshelves-256541.jpg';
import sliderimage3 from '../../media/john-michael-thomson-555276-unsplash.jpg';
import sliderimage4 from '../../media/pj-accetturo-172481-unsplash.jpg';
import sliderimage5 from '../../media/joshua-coleman-623144-unsplash.jpg';
import sliderimage6 from '../../media/radu-marcusu-498248-unsplash.jpg';


/**
 *
 *
 * @export
 * @class Slider
 * @extends {Component}
 */
export default class Slider extends Component {
  
  /**
   * 
   * 
   * @memberof Slider
   * @returns {function} initialises the slider 
   */
  componentDidUpdate() {
    $('.slider').slider({
      full_width: true
    })
  }
  
  
  /**
   * 
   * 
   * @returns {Object} html
   * @memberof Slider 
   */
  render () {
    return (
      <div className="home-page-slider">
        <div className="slider" style={{background: 'rgba(0,0,0,0.7)'}}>
          <ul className="slides">
            <li>
              <img src={sliderimage1} alt="sliderimage1" style={{ backgroundColor: 'rgba(0,0,0,3)'}}/>
              <div className="caption left-align">
                <h1 style={{'textShadow':'0 1px 0 black',}}><strong>Welcome to HelloBooks</strong></h1>
                <h3 style={{'textShadow':'0 1px 0 black'}} className="light grey-text text-lighten-3"><strong>Sign Up today to enter into <br /> a whole book world</strong></h3>
              </div>
            </li>
            <li>
              <img src={sliderimage2} alt=""style={{ backgroundColor: 'rgba(0,0,0,3)'}} />
              <div className="caption left-align">
                <h1 style={{ 'textShadow': '0 1px 0 black', 'color': '#ffffff'}} ><strong>Welcome to the HelloBooks</strong></h1>
                <h3 className="light grey-text text-lighten-3"><strong>You can borrow a book to read <br /> and post reviews on it</strong></h3>
              </div>
            </li>
            <li>
              <img src={sliderimage3} alt=""style={{ backgroundColor: 'rgba(0,0,0,3)'}} />
              <div className="caption left-align">
                <h1 style={{ 'textShadow': '0 1px 0 black', 'color': '#ffffff'}}><strong>Welcome to HelloBooks</strong></h1>
                <h3 className="light grey-text text-lighten-3"><strong>Create Reading list and <br />  suggest books to the library</strong></h3>
              </div>
            </li>
            <li>
              <img src={sliderimage4} alt="" style={{ backgroundColor: 'rgba(0,0,0,3)'}}/>
              <div className="caption left-align">
                <h1 style={{ 'textShadow': '0 1px 0 black', 'color': '#ffffff'}}><strong>Welcome to HelloBooks</strong></h1>
                <h3 className="light grey-text text-lighten-3"><strong>Review books and add <br />books to your favorite list</strong></h3>
              </div>
            </li>
            <li>
              <img src={sliderimage5} alt="" style={{ backgroundColor: 'rgba(0,0,0,3)'}}/>
              <div className="caption left-align">
                <h1 style={{ 'textShadow': '0 1px 0 black', 'color': '#ffffff'}}><strong>Welcome to HelloBooks</strong></h1>
                <h3 className="light grey-text text-lighten-3"><strong>Sign Up today to enter into a whole book world</strong></h3>
              </div>
            </li>
            <li>
              <img src={sliderimage6} alt="" style={{ backgroundColor: 'rgba(0,0,0,3)'}}/>
              <div className="caption left-align">
                <h1 style={{ 'textShadow': '0 1px 0 black', 'color': '#ffffff'}}><strong>Welcome to HelloBooks</strong></h1>
                <h3 className="light grey-text text-lighten-3"><strong>Sign Up today to enter into a whole book world</strong></h3>
              </div>
            </li>

          </ul>
        </div>
      </div>
    )
  }

}