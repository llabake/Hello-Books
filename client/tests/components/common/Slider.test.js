import React from 'react';

import Slider from '../../../components/common/Slider';
import userData from '../../mocks/userData'

test('Slider component renders correctly', () => {
  const wrapper = shallow(<Slider />);
  expect(wrapper).toBeDefined();
})

test('Carousel should call componentDidUpdate', () => {
  let spy;
  afterEach(() => {
    spy.mockClear();
  });

  const props = {
    authenticated: false,
    user: {},
  };
  const nextProps = {
    authenticated: true,
    user: userData.user
  };
  spy = jest.spyOn(Slider.prototype, 'componentDidUpdate')
  const wrapper = shallow(<Slider {...props} />)
  wrapper.setProps(nextProps);
  expect(Slider.prototype.componentDidUpdate).toHaveBeenCalledTimes(1);
  expect(spy).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
})