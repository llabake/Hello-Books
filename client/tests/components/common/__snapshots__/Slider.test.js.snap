// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should call componentDidUpdate 1`] = `
<div
  className="home-page-slider"
>
  <div
    className="slider"
    style={
      Object {
        "background": "rgba(0,0,0,0.7)",
      }
    }
  >
    <ul
      className="slides"
    >
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h1
            style={
              Object {
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to HelloBooks
          </h1>
          <h3
            className="light grey-text text-lighten-3"
            style={
              Object {
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Sign Up today to enter into 
            <br />
             a whole book world
          </h3>
        </div>
      </li>
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h1
            style={
              Object {
                "color": "#ffffff",
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to the HelloBooks
          </h1>
          <h2
            className="light grey-text text-lighten-3"
          >
            You can borrow a book to read 
            <br />
             and post reviews on it
          </h2>
        </div>
      </li>
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h3
            style={
              Object {
                "color": "#ffffff",
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to HelloBooks
          </h3>
          <h5
            className="light grey-text text-lighten-3"
          >
            Create Reading list and 
            <br />
              suggest books to the library
          </h5>
        </div>
      </li>
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h3
            style={
              Object {
                "color": "#ffffff",
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to HelloBooks
          </h3>
          <h5
            className="light grey-text text-lighten-3"
          >
            Review books and add 
            <br />
            books to your favorite list
          </h5>
        </div>
      </li>
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h3
            style={
              Object {
                "color": "#ffffff",
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to HelloBooks
          </h3>
          <h5
            className="light grey-text text-lighten-3"
          >
            Sign Up today to enter into a whole book world
          </h5>
        </div>
      </li>
      <li>
        <img
          alt=""
          src={null}
        />
        <div
          className="caption left-align"
        >
          <h3
            style={
              Object {
                "color": "#ffffff",
                "textShadow": "0 1px 0 black",
              }
            }
          >
            Welcome to HelloBooks
          </h3>
          <h5
            className="light grey-text text-lighten-3"
          >
            Sign Up today to enter into a whole book world
          </h5>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
